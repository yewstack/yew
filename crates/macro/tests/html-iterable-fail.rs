use yew_macro::{html, test_html, test_html_block};

test_html! { |t1| for }
test_html! { |t2| for () }
test_html! { |t3| for {()} }
test_html! { |t4| for Vec::<()>::new().into_iter() }

test_html_block! { |t10|
    let empty = Vec::<()>::new().into_iter();
    html! { for empty }
}

test_html_block! { |t11|
    let empty = Vec::<()>::new();
    html! { for empty.iter() }
}

fn main() {}
