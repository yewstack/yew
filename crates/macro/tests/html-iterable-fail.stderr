error: expected expression after `for`
 --> $DIR/html-iterable-fail.rs:3:19
  |
3 | test_html! { |t1| for }
  |                   ^^^

error[E0277]: `()` is not an iterator
 --> $DIR/html-iterable-fail.rs:4:23
  |
4 | test_html! { |t2| for () }
  |                       ^^ `()` is not an iterator
  |
  = help: the trait `std::iter::Iterator` is not implemented for `()`
  = note: required for the cast to the object type `dyn std::iter::Iterator<Item = _>`

error[E0277]: `()` is not an iterator
 --> $DIR/html-iterable-fail.rs:5:23
  |
5 | test_html! { |t3| for {()} }
  |                       ^^^^ `()` is not an iterator
  |
  = help: the trait `std::iter::Iterator` is not implemented for `()`
  = note: required for the cast to the object type `dyn std::iter::Iterator<Item = _>`

error[E0277]: `()` doesn't implement `std::fmt::Display`
 --> $DIR/html-iterable-fail.rs:6:23
  |
6 | test_html! { |t4| for Vec::<()>::new().into_iter() }
  |                       ^^^ `()` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `()`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
  = note: required because of the requirements on the impl of `std::string::ToString` for `()`
  = note: required because of the requirements on the impl of `std::convert::From<()>` for `yew_shared::virtual_dom::vnode::VNode<_>`
  = note: required because of the requirements on the impl of `std::convert::Into<yew_shared::virtual_dom::vnode::VNode<_>>` for `()`

error[E0277]: `()` doesn't implement `std::fmt::Display`
  --> $DIR/html-iterable-fail.rs:10:17
   |
10 |     html! { for empty }
   |                 ^^^^^ `()` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `()`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: required because of the requirements on the impl of `std::string::ToString` for `()`
   = note: required because of the requirements on the impl of `std::convert::From<()>` for `yew_shared::virtual_dom::vnode::VNode<_>`
   = note: required because of the requirements on the impl of `std::convert::Into<yew_shared::virtual_dom::vnode::VNode<_>>` for `()`

error[E0277]: `()` doesn't implement `std::fmt::Display`
  --> $DIR/html-iterable-fail.rs:15:17
   |
15 |     html! { for empty.iter() }
   |                 ^^^^^ `()` cannot be formatted with the default formatter
   |
   = help: the trait `std::fmt::Display` is not implemented for `()`
   = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
   = note: required because of the requirements on the impl of `std::fmt::Display` for `&()`
   = note: required because of the requirements on the impl of `std::string::ToString` for `&()`
   = note: required because of the requirements on the impl of `std::convert::From<&()>` for `yew_shared::virtual_dom::vnode::VNode<_>`
   = note: required because of the requirements on the impl of `std::convert::Into<yew_shared::virtual_dom::vnode::VNode<_>>` for `&()`

For more information about this error, try `rustc --explain E0277`.
